1. #Task 1 - Fitbit Dataset

1. Read this dataset in pandas 

In [1]:
import pandas as pn

In [44]:
df1 = pn.read_csv(r"C:\Users\dchakrab.COMPUTACENTER\Downloads\FitBit_data.csv")

In [45]:
df1

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0    1503960366    3/25/2016       11004       7.110000         7.110000   
1    1503960366    3/26/2016       17609      11.550000        11.550000   
2    1503960366    3/27/2016       12736       8.530000         8.530000   
3    1503960366    3/28/2016       13231       8.930000         8.930000   
4    1503960366    3/29/2016       12041       7.850000         7.850000   
..          ...          ...         ...            ...              ...   
452  8877689391   04/08/2016       23014      20.389999        20.389999   
453  8877689391   04/09/2016       16470       8.070000         8.070000   
454  8877689391   04/10/2016       28497      27.530001        27.530001   
455  8877689391   04/11/2016       10622       8.060000         8.060000   
456  8877689391   04/12/2016        2350       1.780000         1.780000   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                         0.0                2.57                      0.46   
1                         0.0                6.92                      0.73   
2                         0.0                4.66                      0.16   
3                         0.0                3.19                      0.79   
4                         0.0                2.16                      1.09   
..                        ...                 ...                       ...   
452                       0.0               11.10                      0.63   
453                       0.0                0.00                      0.02   
454                       0.0               21.92                      1.12   
455                       0.0                1.47                      0.15   
456                       0.0                0.00                      0.00   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                   4.07                     0.00                 33   
1                   3.91                     0.00                 89   
2                   3.71                     0.00                 56   
3                   4.95                     0.00                 39   
4                   4.61                     0.00                 28   
..                   ...                      ...                ...   
452                 8.62                     0.00                 70   
453                 8.02                     0.00                 90   
454                 4.46                     0.00                128   
455                 6.37                     0.01                 18   
456                 1.78                     0.00                  0   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                     12                   205               804      1819  
1                     17                   274               588      2154  
2                      5                   268               605      1944  
3                     20                   224              1080      1932  
4                     28                   243               763      1886  
..                   ...                   ...               ...       ...  
452                   29                   359               982      4196  
453                    9                   289              1052      3841  
454                   46                   211              1055      4526  
455                    7                   225              1190      2820  
456                    0                    58               531       938  

[457 rows x 15 columns]

df1.dtypes

*********load dataset in pymongo***********

In [14]:
import pymongo as py

In [16]:
client = py.MongoClient("mongodb+srv://root:mongodb123@cluster0.zlbwi.mongodb.net/?retryWrites=true&w=majority")
db = client.test

In [17]:
df1.to_json("FitbitData.json")

In [20]:
import json

In [21]:
with open("FitbitData.json") as file:
    file_data=json.load(file)
database = client['FitBitData']
collection = database['FitBitData_table']
collection.insert_one(file_data)

convert all the dates avaible in dataset to timestamp format in pandas

In [67]:
df1['converted_activity_date'] = pn.to_datetime(df1['ActivityDate'])

In [24]:
df1.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    3/25/2016       11004           7.11             7.11   
1  1503960366    3/26/2016       17609          11.55            11.55   
2  1503960366    3/27/2016       12736           8.53             8.53   
3  1503960366    3/28/2016       13231           8.93             8.93   
4  1503960366    3/29/2016       12041           7.85             7.85   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                2.57                      0.46   
1                       0.0                6.92                      0.73   
2                       0.0                4.66                      0.16   
3                       0.0                3.19                      0.79   
4                       0.0                2.16                      1.09   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 4.07                      0.0                 33   
1                 3.91                      0.0                 89   
2                 3.71                      0.0                 56   
3                 4.95                      0.0                 39   
4                 4.61                      0.0                 28   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
0                   12                   205               804      1819   
1                   17                   274               588      2154   
2                    5                   268               605      1944   
3                   20                   224              1080      1932   
4                   28                   243               763      1886   

  converted_activity_date  
0              2016-03-25  
1              2016-03-26  
2              2016-03-27  
3              2016-03-28  
4              2016-03-29

In [ ]:
>>>>>>>>>>>>Find out in data that how many unique id we have>>>>>>>>>>>>>>>>>>

In [25]:
pn.unique(df1['Id'])

array([1503960366, 1624580081, 1644430081, 1844505072, 1927972279,
       2022484408, 2026352035, 2320127002, 2347167796, 2873212765,
       2891001357, 3372868164, 3977333714, 4020332650, 4057192912,
       4319703577, 4388161847, 4445114986, 4558609924, 4702921684,
       5553957443, 5577150313, 6117666160, 6290855005, 6391747486,
       6775888955, 6962181067, 7007744171, 7086361926, 8053475328,
       8253242879, 8378563200, 8583815059, 8792009665, 8877689391],
      dtype=int64)

In [26]:
len(pn.unique(df1['Id']))

35

In [ ]:
>>>>>>>>>>which id is one of the active id that you have in whole dataset>>>>>>>>>>

In [56]:
df1[df1['VeryActiveMinutes']== max(df1['VeryActiveMinutes'])]['Id']

304    5577150313
Name: Id, dtype: int64

In [ ]:
>>>>>>>>>>>>>>>>>>>How many of them have not logged there activity find out in terms of number of ids>>>>>>>>>>>>>>>>

In [36]:
df2 = df1[df1['LoggedActivitiesDistance'] == 0]

In [38]:
pn.unique(df2['Id'])

array([1503960366, 1624580081, 1644430081, 1844505072, 1927972279,
       2022484408, 2026352035, 2320127002, 2347167796, 2873212765,
       2891001357, 3372868164, 3977333714, 4020332650, 4057192912,
       4319703577, 4388161847, 4445114986, 4558609924, 4702921684,
       5553957443, 5577150313, 6117666160, 6290855005, 6391747486,
       6775888955, 6962181067, 7007744171, 7086361926, 8053475328,
       8253242879, 8378563200, 8583815059, 8792009665, 8877689391],
      dtype=int64)

In [39]:
len(pn.unique(df2['Id']))

35

In [103]:
df1.groupby('Id')['VeryActiveMinutes'].max()

Id
1503960366     89
1624580081      7
1644430081     59
1844505072      9
1927972279      0
2022484408     72
2026352035      0
2320127002     10
2347167796     51
2873212765     46
2891001357      0
3372868164     24
3977333714     35
4020332650     31
4057192912     30
4319703577     15
4388161847      0
4445114986     28
4558609924     27
4702921684     15
5553957443     51
5577150313    202
6117666160      3
6290855005     33
6391747486     46
6775888955     56
6962181067     73
7007744171    104
7086361926     76
8053475328    165
8253242879     47
8378563200    113
8583815059      5
8792009665      9
8877689391    128
Name: VeryActiveMinutes, dtype: int64

In [ ]:
>>>>>>>>>>>>>>>>>>>>>>>Find out who is the laziest person id that we have in dataset>>>>>>>>>>>>>

In [105]:
df1.groupby('Id')['SedentaryMinutes'].max().nlargest(5)

Id
1844505072    1440
2320127002    1440
2873212765    1440
2891001357    1440
4020332650    1440
Name: SedentaryMinutes, dtype: int64

In [ ]:
>>>>>>>>>Explore over an internet that how much calories burn is required for a healthy person and find out how many healthy person we have in our dataset>>>> 

In [61]:
len(pn.unique(df1[(df1['Calories']>=2200) & (df1['Calories']<=3000)]['Id']))

27

In [ ]:
>>>>>>27 people are healthy in terms of daily calorie burn>>>>>>

In [68]:

df1

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0    1503960366    3/25/2016       11004       7.110000         7.110000   
1    1503960366    3/26/2016       17609      11.550000        11.550000   
2    1503960366    3/27/2016       12736       8.530000         8.530000   
3    1503960366    3/28/2016       13231       8.930000         8.930000   
4    1503960366    3/29/2016       12041       7.850000         7.850000   
..          ...          ...         ...            ...              ...   
452  8877689391   04/08/2016       23014      20.389999        20.389999   
453  8877689391   04/09/2016       16470       8.070000         8.070000   
454  8877689391   04/10/2016       28497      27.530001        27.530001   
455  8877689391   04/11/2016       10622       8.060000         8.060000   
456  8877689391   04/12/2016        2350       1.780000         1.780000   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                         0.0                2.57                      0.46   
1                         0.0                6.92                      0.73   
2                         0.0                4.66                      0.16   
3                         0.0                3.19                      0.79   
4                         0.0                2.16                      1.09   
..                        ...                 ...                       ...   
452                       0.0               11.10                      0.63   
453                       0.0                0.00                      0.02   
454                       0.0               21.92                      1.12   
455                       0.0                1.47                      0.15   
456                       0.0                0.00                      0.00   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                   4.07                     0.00                 33   
1                   3.91                     0.00                 89   
2                   3.71                     0.00                 56   
3                   4.95                     0.00                 39   
4                   4.61                     0.00                 28   
..                   ...                      ...                ...   
452                 8.62                     0.00                 70   
453                 8.02                     0.00                 90   
454                 4.46                     0.00                128   
455                 6.37                     0.01                 18   
456                 1.78                     0.00                  0   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
0                     12                   205               804      1819   
1                     17                   274               588      2154   
2                      5                   268               605      1944   
3                     20                   224              1080      1932   
4                     28                   243               763      1886   
..                   ...                   ...               ...       ...   
452                   29                   359               982      4196   
453                    9                   289              1052      3841   
454                   46                   211              1055      4526   
455                    7                   225              1190      2820   
456                    0                    58               531       938   

    converted_activity_date  
0                2016-03-25  
1                2016-03-26  
2                2016-03-27  
3                2016-03-28  
4                2016-03-29  
..                      ...  
452              2016-04-08  
453              2016-04-09  
454              2016-04-10  
455              2016-04-11  
456              20

In [84]:
df1['converted_activity_week']=df1['converted_activity_date'].dt.week

C:\Users\dchakrab.COMPUTACENTER\AppData\Local\Temp\ipykernel_27064\1326656825.py:1: FutureWarning: Series.dt.weekofyear and Series.dt.week have been deprecated. Please use Series.dt.isocalendar().week instead.
  df1['converted_activity_week']=df1['converted_activity_date'].dt.week


In [85]:
df1.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    3/25/2016       11004           7.11             7.11   
1  1503960366    3/26/2016       17609          11.55            11.55   
2  1503960366    3/27/2016       12736           8.53             8.53   
3  1503960366    3/28/2016       13231           8.93             8.93   
4  1503960366    3/29/2016       12041           7.85             7.85   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                2.57                      0.46   
1                       0.0                6.92                      0.73   
2                       0.0                4.66                      0.16   
3                       0.0                3.19                      0.79   
4                       0.0                2.16                      1.09   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 4.07                      0.0                 33   
1                 3.91                      0.0                 89   
2                 3.71                      0.0                 56   
3                 4.95                      0.0                 39   
4                 4.61                      0.0                 28   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
0                   12                   205               804      1819   
1                   17                   274               588      2154   
2                    5                   268               605      1944   
3                   20                   224              1080      1932   
4                   28                   243               763      1886   

  converted_activity_date  converted_activity_month  converted_activity_week  
0              2016-03-25                        12                       12  
1              2016-03-26                        12                       12  
2              2016-03-27                        12                       12  
3              2016-03-28                        13                       13  
4              2016-03-29                        13                       13

In [ ]:
how many person are not a regular person with respect to activity try to find out those.

In [101]:
df1.groupby('Id')['converted_activity_week'].count().nsmallest()

Id
2891001357    8
4388161847    8
8583815059    8
6391747486    9
6775888955    9
Name: converted_activity_week, dtype: int64

From the above series we can easliy identify that Ids 2891001357, 4388161847, 6391747486,6775888955 and 8583815059 are not regular active person

In [ ]:
>>>>>>>>>>who is the third most active person in this dataset find out those in pandas and in sql both>>>>>>>>>>>>>>

In [96]:
df1.groupby('Id')['VeryActiveMinutes'].sum().nlargest(3)

Id
8053475328    933
5577150313    900
8877689391    802
Name: VeryActiveMinutes, dtype: int64

In [ ]:
>>>>>>>>>>>>Who is the 5th most laziest person avilable in dataset find it out>>>>>>>>>>

In [97]:
df1.groupby('Id')['SedentaryMinutes'].sum().nlargest(5)

Id
4057192912    43458
4020332650    34663
1624580081    24288
8253242879    16228
1503960366    15387
Name: SedentaryMinutes, dtype: int64

In [ ]:
>>>>>>>>>>>>>>>>>>>>> what is a total acumulative calories burn for a person find out.>>>>>>>>>>>>>>>>>>

In [106]:
df1.groupby(['Id','converted_activity_week'])['Calories'].sum()

Id          converted_activity_week
1503960366  12                          5917
            13                         13088
            14                         13262
            15                          1861
1624580081  12                          3987
                                       ...  
8792009665  14                         14500
            15                          2453
8877689391  13                         11611
            14                         26045
            15                          3758
Name: Calories, Length: 103, dtype: int64